AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOperation.swift

```diff
@@ -208,6 +223,46 @@ final class InitialSyncOperation: AsynchronousOperation {
         super.finish()
     }
 
+    private func onReceiveMutationEvent(value: IncomingEventReconciliationQueueEvent) {
+        switch value {
+        case .mutationEvent(let event):
+            guard event.modelName == modelType.modelName else {
+                return
+            }
+            reconciledReceived += 1
```

Comment:
I don't think the count is a brittle way since no better way I can think of so far. But the race condition do need to be solved (perhaps using semaphore?)